internal/openmetrics-exporter/drive_collector.go

```diff
@@ -0,0 +1,47 @@
+package collectors
+
+
+import (
+	"github.com/prometheus/client_golang/prometheus"
+	"purestorage/fa-openmetrics-exporter/internal/rest-client"
+)
+
+type DriveCollector struct {
+	CapacityDesc     *prometheus.Desc
+	StatusDesc       *prometheus.Desc
+	TypeDesc         *prometheus.Desc
+        NameDesc         *prometheus.Desc
+        ProtocolDesc     *prometheus.Desc
+	Client           *client.FAClient
+}
+
+func (c *DriveCollector) Describe(ch chan<- *prometheus.Desc) {
+	prometheus.DescribeByCollect(c, ch)
+}
+
+func (c *DriveCollector) Collect(ch chan<- prometheus.Metric) {
+	drl := c.Client.GetDrives()
+	if len(drl.Items) == 0 {
+		return
+	}
+	for _, d := range drl.Items {
+		ch <- prometheus.MustNewConstMetric(
+			c.CapacityDesc,
+			prometheus.GaugeValue,
+			float64(d.Capacity),
+			d.Name, d.Type, d.Status, d.Protocol,
+		)
+		}
+}
+
+func NewDriveCollector(fa *client.FAClient) *DriveCollector {
+	return &DriveCollector{
+		StatusDesc: prometheus.NewDesc(
+			"purefa_drive_capacity",
```

Comment:
per openmetrics standards, the unit should be in the metric name and the description. I believe these are bytes - and update in `purefa-metrics.md`